app/src/main/java/zapsolutions/zap/connection/establishConnectionToLnd/LndConnection.java

```diff
@@ -49,45 +54,26 @@ public static synchronized LndConnection getInstance() {
 
     private void readSavedConnectionInfo() {
 
-        App ctx = App.getAppContext();
+        // Load current wallet connection config
+        mConnectionConfig = WalletConfigsManager.getInstance().loadCurrentWalletConfig();
 
-        mPrefsRemote = Armadillo.create(ctx, PrefsUtil.PREFS_REMOTE)
-                .encryptionFingerprint(ctx)
-                .keyStretchingFunction(new PBKDF2KeyStretcher(RefConstants.NUM_HASH_ITERATIONS, null))
-                .password(ctx.inMemoryPin.toCharArray())
-                .contentKeyDigest(UtilFunctions.getZapsalt().getBytes())
-                .build();
+        // Generate Macaroon
+        mMacaroon = new MacaroonCallCredential(mConnectionConfig.getMacaroon());
 
-        // The following string contains host,port,cert and macaroon in one string separated with ";"
-        // This way we can read all necessary data in one call and do not have to execute the key stretching function 4 times.
-        String connectionInfo = mPrefsRemote.getString(PrefsUtil.REMOTE_COMBINED, "");
-        mConnectionInfo = connectionInfo.split(";");
-        ZapLog.debug(LOG_TAG, connectionInfo);
+        mSSLFactory = null;
```

Comment:
Yes, but if the wallet is switched (in the future) or connection resetted in the settings (not ResetAll) then mSSLFactory is still set even if your new connection is a BTCPay connection without a certificate. It would then try to connect using the old certificate and fail.